rllib/env/single_agent_env_runner.py

```diff
@@ -637,15 +636,16 @@ def make_env(self) -> None:
                 env_context=env_ctx,
             )
         gym.register("rllib-single-agent-env-v0", entry_point=entry_point)
+        vectorize_mode = self.config.gym_env_vectorize_mode
 
         self.env = DictInfoToList(
             gym.make_vec(
                 "rllib-single-agent-env-v0",
                 num_envs=self.config.num_envs_per_env_runner,
                 vectorization_mode=(
-                    VectorizeMode.ASYNC
-                    if self.config.remote_worker_envs
-                    else VectorizeMode.SYNC
+                    vectorize_mode
+                    if isinstance(vectorize_mode, gym.envs.registration.VectorizeMode)
```

Comment:
Isn't this already validated in `AlgorithmConfig`? Imo we need to validate only once. Either here or in the `AlgorithmConfig`.

Reply:
True, but an AlgorithmConfig only gets validated in the algo, not when it enters an EnvRunner. So - in theory - users could still run into this if they instantiate an EnvRunner in isolation with some config that has this defined as a string (e.g. "SYNC").